jellyfin/jellyfin
# Fix replace logic

We never passed the flag for metadata replacement to the remote provider logic, hence the remote result would never replace what's already there.
Additionally, the logic did not repect if an item was locked or not


**Changes**
* Pass replace flag to remote providers
* Fix merge logic to only override empty fields, except when merging with the actual item
  * this will gradually build up metadata beginning with local providers and going through remote providers later on

**Issues**
Fixes #8532 (potentially)
Fixes #11294 (the part where it is repopulated)
Fixes #11656
Fixes #11717
Fixes #11773 (potentially)
Fixes #11831

**Testing**
* Add series, manually remove some metadata and replace all metadata -> should refill
* Add series, manually remove some metadata and fetch missing metadata -> should refill
* Add series, manually identify, remove some metadata and replace metadata -> should keep identified series
* Any other combination of the above and with replace images
